nixos/modules/services/network-filesystems/glusterfs.nix

```diff
@@ -41,6 +41,53 @@ in
         default = "INFO";
       };
 
+      useRpcbind = mkOption {
+        type = types.bool;
+        description = ''
+          Enable use of rpcbind. This is required for Gluster's NFS functionality.
+
+          You may want to turn it off to reduce the attack surface for DDoS reflection attacks.
+
+          See https://davelozier.com/glusterfs-and-rpcbind-portmap-ddos-reflection-attacks/
+          and https://bugzilla.redhat.com/show_bug.cgi?id=1426842 for details.
+        '';
+        default = true;
+      };
+
+      enableGlustereventsd = mkOption {
+        type = types.bool;
+        description = "Whether to enable the GlusterFS Events Daemon";
+        default = true;
+      };
+
+      killMode = mkOption {
+        type = types.enum ["control-group" "process" "mixed" "none"];
+        description = ''
+          The systemd KillMode to use for glusterd.
+
+          glusterd spawns other daemons like gsyncd.
+          If you want these to stop when glusterd is stopped (e.g. to ensure
+          that NixOS config changes are reflected even for these sub-daemons),
+          set this to 'control-group'.
+        '';
+        default = "process";
+      };
```

Comment:
Does this need to be configurable? If yes give a quick description in which cases the different modes might be useful.

Reply:
@bachp Updated; I have now described both typical use cases.

I have also added a commit that changes the default to `control-group`, as you and me both think that this is the better default (from https://github.com/NixOS/nixpkgs/pull/27340#discussion_r139311268).